rust/connlib/tunnel/src/lib.rs

```diff
@@ -295,23 +296,46 @@ where
             resources_gateways,
             ice_candidate_queue,
             callbacks: CallbackErrorFacade(callbacks),
+            iface_handler_abort,
         })
     }
 
+    #[tracing::instrument(level = "trace", skip(self))]
+    pub async fn add_route(self: &Arc<Self>, route: IpNetwork) -> Result<()> {
+        let mut device_io = self.device_io.write().await;
+        let mut iface_config = self.iface_config.write().await;
```

Comment:
If we do these two in the wrong order anywhere, we might run into a deadlock. Can't we combine them into one structure?